pkg/id/mod.ts

```diff
@@ -67,3 +68,22 @@ export class SnowflakeIDGenerator {
     return Result.ok(id.toString() as ID<T>);
   }
 }
+
+export const IDSchema = <T>() =>
+  z
+    .string()
+    .regex(/^\d+$/)
+    .refine((s) => {
+      let n;
+      try {
+        n = BigInt(s);
+      } catch {
+        return false;
+      }
+
+      return (
+        n >= 0b1111111111111111111111n &&
+        n <= 0b1111111111111111111111111111111111111111111111111111111111111111n
+      );
```

Comment:
```suggestion
      try {
        const n = BigInt(s);
        return (
          n >= 0b1111111111111111111111n &&
          n <= 0b1111111111111111111111111111111111111111111111111111111111111111n
        );
      } catch {
        return false;
      }
```
無理に `let` で外に出さなくても, これでどうでしょう

Reply:
3e8560162f51905bcbbad6e8a4a7e5874b59c113 で対応しました.